pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -173,6 +91,153 @@ func registerLogicalDataReplicationTests(r registry.Registry) {
+		debugRunDuration:        10 * time.Minute,
+		maxBlockBytes:           1024,
+		initWithSplitAndScatter: true}
+
+	leftJobID, rightJobID := setupLDR(ctx, t, c, setup, kvWorkload)
+
+	// Setup latency verifiers, remembering to account for latency spike from killing a node
+	maxExpectedLatency := 5 * time.Minute
+	llv := makeLatencyVerifier("ldr-left", 0, maxExpectedLatency, t.L(),
+		getLogicalDataReplicationJobInfo, t.Status, false /* tolerateErrors */)
+	defer llv.maybeLogLatencyHist()
+
+	rlv := makeLatencyVerifier("ldr-right", 0, maxExpectedLatency, t.L(),
+		getLogicalDataReplicationJobInfo, t.Status, false /* tolerateErrors */)
+	defer rlv.maybeLogLatencyHist()
+
+	workloadDoneCh := make(chan struct{})
+	debugZipFetcher := &sync.Once{}
+	getDebugZips := func() {
+		if err := c.FetchDebugZip(ctx, t.L(), "latency_left_debug.zip", setup.left.nodes); err != nil {
+			t.L().Errorf("could not fetch debug zip: %v", err)
+		}
+		if err := c.FetchDebugZip(ctx, t.L(), "latency_right_debug.zip", setup.right.nodes); err != nil {
+			t.L().Errorf("could not fetch debug zip: %v", err)
+		}
+	}
+
+	monitor := c.NewMonitor(ctx, setup.CRDBNodes())
+	monitor.Go(func(ctx context.Context) error {
+		if err := llv.pollLatencyUntilJobSucceeds(ctx, setup.left.db, leftJobID, time.Second, workloadDoneCh); err != nil {
+			debugZipFetcher.Do(getDebugZips)
+			return err
+		}
+		return nil
+	})
+	monitor.Go(func(ctx context.Context) error {
+		if err := rlv.pollLatencyUntilJobSucceeds(ctx, setup.right.db, rightJobID, time.Second, workloadDoneCh); err != nil {
+			debugZipFetcher.Do(getDebugZips)
+			return err
+		}
+		return nil
+	})
+
+	monitor.Go(func(ctx context.Context) error {
+		defer close(workloadDoneCh)
+		return c.RunE(ctx, option.WithNodes(setup.workloadNode), kvWorkload.sourceRunCmd("system", setup.CRDBNodes()))
+	})
+
+	// Let workload run for a bit before we kill a node
+	time.Sleep(time.Minute * 1)
+
+	findNodeToStop := func(info *clusterInfo, rng *rand.Rand) int {
+		for {
+			anotherNode := info.nodes.SeededRandNode(rng)[0]
+			if anotherNode != info.gatewayNodes[0] {
+				return anotherNode
+			}
+		}
+	}
+
+	t.L().Printf("Finding node to stop Left")
+	nodeToStopL := findNodeToStop(setup.left, setup.rng)
+	t.L().Printf("Finding node to stop right")
+	nodeToStopR := findNodeToStop(setup.right, setup.rng)
+
+	// Graceful shutdown on both nodes
+	// TODO(naveen.setlur): maybe switch this to a less graceful shutdown via SIGKILL
+	t.L().Printf("Shutting down node-left: %d", nodeToStopL)
+	monitor.ExpectDeath()
+	if err := c.StopCockroachGracefullyOnNode(ctx, t.L(), nodeToStopL); err != nil {
+		t.Fatalf("Unable to shutdown node: %s", err)
+	}
+
+	t.L().Printf("Shutting down node-right: %d", nodeToStopR)
+	monitor.ExpectDeath()
+	if err := c.StopCockroachGracefullyOnNode(ctx, t.L(), nodeToStopR); err != nil {
+		t.Fatalf("Unable to shutdown node: %s", err)
+	}
+
+	monitor.Wait()
+	VerifyCorrectness(t, setup, leftJobID, rightJobID, 5*time.Minute)
```

Comment:
to double check: do the logs indicate that the workload runs for like 15 minutes before `VerifyCorrectness` gets called?

Reply:
Yup, duration is currently 10 mins and that's what I'm seeing in stdout